### PR TITLE
Use `pypi/` instead of `python/` for Python package repos

### PR DESCRIPTION
**Pull request by olafurpg**
Following up from
https://sourcegraph.slack.com/archives/C032G1CLM7X/p1652957835148599.
The `python/` prefix is undesirable because makes it confusing
to know the difference between `repo:^python` and `lang:python`.
With `python/` as the prefix it's natural to use the phrase "python repository"
to explain such repos but that can quickly be confused any regular repository
that has Python code. The "pypi/" prefix makes it clearer that these are 
python packages only.

For consistency with the `npm/` and `maven/` prefixes, the `pypi/` prefix
works better than `python/`.

cc/ @tjdevries 

## Test plan

See the CI go green.

### CODE REVIEW

File: internal/conf/reposource/python_packages.go [hunk 1]
```diff
@@ -38,15 +38,15 @@ func ParsePythonDependency(dependency string) (*PythonDependency, error) {
 // ParsePythonDependencyFromRepoName is a convenience function to parse a repo name in a
 // 'python/<name>(==<version>)?' format into a PythonDependency.
 func ParsePythonDependencyFromRepoName(name string) (*PythonDependency, error) {
-	dependency := strings.TrimPrefix(name, "python/")
+	dependency := strings.TrimPrefix(name, "pip/")
 	if len(dependency) == len(name) {
 		return nil, errors.New("invalid python dependency repo name, missing python/ prefix")
```
Comment (stefanhengl) [inline internal/conf/reposource/python_packages.go:43]:
```suggestion
		return nil, errors.New("invalid python dependency repo name, missing pip/ prefix")
```

**Review by stefanhengl — COMMENTED** [review]
I think we need to add a migration, don't we?

File: internal/conf/reposource/python_packages.go (hunk 1 above)
Comment (olafurpg) [reply internal/conf/reposource/python_packages.go]: Good catch. I updated references to `python/` in various places based on this suggestion.

**Comment by olafurpg** [comment]
> I think we need to add a migration, don't we?

Correct. I think we need to update the names of existing repos with the `python/` prefix. I preserved the "python" name in the scheme because it's not user-facing. We similarly use "jvm" as the underlying scheme while we use `maven/` as the URL prefix.

**Review by unknwon — APPROVED** [review]

**Comment by olafurpg** [comment]
Closing this PR in favor of the approach that was approved in RFC 698

**Pull request closed without merging**